doc/input.md

```diff
@@ -103,9 +103,18 @@ class MyGame extends Game with TapDetector {
 ```
 You can also check a more complete example [here](/doc/examples/gestures).
 
-## Tapable components
+## Tapable and Dragable components
 
-Flame also offers a simple helper to make it easier to handle tap events on `PositionComponent`. By adding the `HasTapableComponents` mixin to your game, and using the mixin `Tapable` on your components can override the following methods, enabling easy to use tap events on yours components.
+A component can extend the `Tapable` and/or the `Dragable` mixins to handle taps and drags on the component.
+
+All overridden methods return a boolean to control if the event should be passed down further along to components underneath it.
+So say that you only want your top visible component to receive a tap and not the ones underneath it, then your `onTapDown`,  `onTapUp` and `onTapCancel` implementations should return `false`, if you want the event to go through more of the components underneath then you should return `true`.
+
+The same applies if your component has children, then the event is first sent to the leaves in the children tree and then passed further down until a method returns `false`.
+
+### Tapable components
+
+Flame also offers a simple helper to make it easier to handle tap events on `PositionComponent` (and `BaseComponent`). By adding the `HasTapableComponents` mixin to your game, and using the mixin `Tapable` on your components can override the following methods, enabling easy to use tap events on your components.
```

Comment:
The first sentence seems to be conflicting with the previous paragraph; we just said that there are Tappable and Draggable as simple helpers to handle events on BaseComponent.

Reply:
Yeah, this needs to be updated.